src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -801,8 +801,7 @@ public virtual IHtmlContent GenerateAntiforgery(ViewContext viewContext)
                 throw new ArgumentNullException(nameof(viewContext));
             }
 
-            var formContext = viewContext.ClientValidationEnabled ? viewContext.FormContext : null;
```

Comment:
This code wasn't really making use of the `FormContext`, discovered this while playing with lazy implementations. Refactoring for clarity.